py-polars/tests/unit/lazyframe/test_optimizations.py

```diff
@@ -0,0 +1,82 @@
+import polars as pl
+from polars.testing import assert_frame_equal
+
+
+def test_is_null_followed_by_all() -> None:
+    lf = pl.LazyFrame({"group": [0, 0, 0, 1], "val": [6, 0, None, None]})
+
+    expected_df = pl.DataFrame({"group": [0, 1], "val": [False, True]})
+    result_df = lf.group_by("group").agg(pl.col("val").is_null().all()).collect()
+
+    assert_frame_equal(expected_df, result_df, check_row_order=False)
```

Comment:
You should also assert here that the expression was replaced in `lf.explain()` - you would assert that the re-written function was not in the explain output, and then assert that the new function is in the explain output.

For an example of this see https://github.com/pola-rs/polars/blob/e3f3651832f988be0a6a0d1ec497af513e951939/py-polars/tests/unit/test_predicates.py#L281-L282